src/framework/uicomponents/view/abstracttoolbarmodel.cpp

```diff
@@ -322,3 +326,19 @@ ToolBarItem& AbstractToolBarModel::item(const ToolBarItemList& items, const Acti
     static ToolBarItem dummy;
     return dummy;
 }
+
+void AbstractToolBarModel::updateShortcutsAll()
+{
+    auto screg = shortcutsRegister();
+
+    for (ToolBarItem* toolBarItem : m_items) {
+        if (!toolBarItem) {
+            continue;
+        }
+
+        UiAction action = toolBarItem->action();
+        action.shortcuts = screg->shortcut(action.code).sequences;
+
+        toolBarItem->setAction(action);
```

Comment:
It looks like it would be necessary to update the ToolbarItem's `menuItems` too. Probably the best way to do that is by creating another method that takes a `std::vector<MenuItem*>` and updates those menu items, and recurses over those items' subitems.